### ISSUE TITLE
[Enhancement] Literal string argument type (ex. /gamemode survival)

### ISSUE THREAD

**Issue by Combustible** [JorelAli/1.13-Command-API#9]
In the FAQ, you have this section: 
```Why is there no support for lists as arguments```

The explanation there makes sense - every argument that can be tab completed like this is another node in the top-level command tree. However...

If you provided some kind of literal argument type (a string - for example "creative" like used in /gamemode) the user could build their own argument list by calling your register() command in a loop. For example, calling register() with /gamemode creative, /gamemode survival, etc. This would make it really obvious to the caller that making a giant list of things (e.g. Material) is a bad idea because they would have to call register each time (at least, I hope they'd figure that out...)

Not super important, but I'd use it if it existed.

**Comment by Combustible** [comment]
Also - it occurs to me we could use this to build nested commands. For example: 

```
/myplugin mycommand1 <args> 
/myplugin mycommand2 <different_args>```

**Comment by JorelAli** [comment]
Doable. I've managed to create a [LiteralArgument class](https://github.com/JorelAli/1.13-Command-API/commit/0a1fcd5597314750278bf907a616a57f15989cb2#diff-cd2f2fd8207847817582d8a37d39bc00) and as of the [latest commit](https://github.com/JorelAli/1.13-Command-API/commit/dbc4ec020941d151dc0bdeb7c96cea1cf98cd443), created a (semi) successful implementation of it. I will fix the implementation at a later date, after I check for bugs and experiment with it a bit more.

The literal implementation is very different to regular arguments and thus, creates more tests to look out for.

> Not super important, but I'd use it if it existed.
Same.

**Comment by JorelAli** [comment]
As of [this commit](https://github.com/JorelAli/1.13-Command-API/commit/06731a2d4e6790194a791c1b12162475a008b4e8), LiteralArguments have been fully implemented. I noticed that during my testing, using `arguments.clear()` to clear my `LinkedHashMap` had a slightly different effect compared to simply creating a new `LinkedHashMap` _(perhaps it was because `clear()` was too slow?)_

I believe my main reason for not implementing this was the danger of abusing it. Of course, any smart developer would realise that it would be inefficient to register a command multiple times for an insanely large list, HOWEVER it can be the icing on the cake for small lists.

If you were wondering, this poor code would create approximately 214,369 commands, take approximately 3 hours, 15 minutes to generate and make a file of about 24MB in size. All to allow the syntax of `/stupidcommand <material> <material>`
```java
//Most retarded code ever incoming:
System.out.println("Installing " + Material.values().length + " commands");
for(Material m : Material.values()) {
	arguments = new LinkedHashMap<>();
	arguments.put("item", new LiteralArgument(m.name()));
	for(Material m2 : Material.values()) {
		arguments.put("seconditem", new LiteralArgument(m2.name()));
		CommandAPI.getInstance().register("stupidcmd", arguments, (sender, args) -> {
			sender.sendMessage("Executed successfully");
		});
	}
}
```

All nonsense aside, look out for the documentation on using literals, I have implemented them very differently to other argument types.
